### ISSUE TITLE
Testing GPU Memory

### ISSUE THREAD

**Issue by lizolson** [feihuzhang/GANet#33]
I have four 12 GB GPUs. I want to predict on higher resolution images (Cityscape, which is 2048x1024), and I understand the width and height have to be multiples of 48 (crop dimensions: 2016 x1008). However I get a memory error when running predict.py, and it seems like it's not fully utilizing multiple GPUs. Is the code set-up to utilize multiple GPUs in predict.py, or do I need to make a modification? Thank you!

**Comment by feihuzhang** [comment]
An easy way, you can simply slice the image, calculate the results and then merge them. Something like follows code. it only takes 7.5G mem. You can set crop_height x crop_width to 1056x2064. It works for me with only some extra errors around the image borders.
 
    with torch.no_grad():
        overlap = 96
        h_slice = 288

        temp0 = model(input1[:,:,0:h_slice,:], input2[:,:,0:h_slice,:])
        temp1 = model(input1[:,:,h_slice-overlap:2*h_slice-overlap,:], input2[:,:,h_slice-overlap:2*h_slice-overlap,:])
        temp2 = model(input1[:,:,2*h_slice-2*overlap:3*h_slice-2*overlap,:], input2[:,:,2*h_slice-2*overlap:3*h_slice-2*overlap,:])
        temp3 = model(input1[:,:,3*h_slice-3*overlap:4*h_slice-3*overlap,:], input2[:,:,3*h_slice-3*overlap:4*h_slice-3*overlap,:])
        temp4 = model(input1[:,:,4*h_slice-4*overlap:5*h_slice-4*overlap,:], input2[:,:,4*h_slice-4*overlap:5*h_slice-4*overlap,:])

    temp = np.zeros([1,opt.crop_height, opt.crop_width])

    temp0 = temp0.cpu()
    temp0 = temp0.detach().numpy()
    temp1 = temp1.cpu()
    temp1 = temp1.detach().numpy()
    temp2 = temp2.cpu()
    temp2 = temp2.detach().numpy()
    temp3 = temp3.cpu()
    temp3 = temp3.detach().numpy()
    temp4 = temp4.cpu()
    temp4 = temp4.detach().numpy()
    temp[0,0:h_slice,:]=temp0[0,:,:]
    temp[0,h_slice-int(overlap/2):2*h_slice-overlap,:]=temp1[0,int(overlap/2):,:]
    temp[0,2*h_slice-2*overlap+int(overlap/2):3*h_slice-2*overlap,:]=temp2[0,int(overlap/2):,:]
    temp[0,3*h_slice-3*overlap+int(overlap/2):4*h_slice-3*overlap,:]=temp3[0,int(overlap/2):,:]
    temp[0,4*h_slice-4*overlap+int(overlap/2):5*h_slice-4*overlap,:]=temp4[0,int(overlap/2):,:]
    
    if height <= opt.crop_height and width <= opt.crop_width:
        temp = temp[0, opt.crop_height - height: opt.crop_height, opt.crop_width - width: opt.crop_width]
    else:
        temp = temp[0, :, :]
    skimage.io.imsave(savename, (temp * 256).astype('uint16'))
